src/client/telemetry/importTracker.ts

```diff
@@ -18,9 +19,14 @@ import { IImportTracker } from './types';
 const ImportRegEx = /^(?!['"#]).*from\s+([a-zA-Z0-9_\.]+)\s+import.*(?!['"])|^(?!['"#]).*import\s+([a-zA-Z0-9_\., ]+).*(?!['"])/;

```

Comment:
```suggestion
const ImportRegEx = /^\s*(?from|import)\s+([^.]+)\s+/;
```
This should capture only absolute imports and only the top-level package which is all we would be interested in.

Reply:
That would miss something like 
import sklearn.foo 

And it would not handle the case where an import was in a string or a comment.

---
In reply to: [267956790](https://github.com/Microsoft/vscode-python/pull/4875#discussion_r267956790) [](ancestors = 267956790)

Reply:
You're right because I forgot to strip out the trailing `\s`. 😄 

As for missing imports in a string or comment, why would we want that? If the imports aren't being used then why would we care about that case? And that should be rare enough of a case that it wouldn't skew learning what the most popular public packages used are.

Reply:
No I meant the opposite, your string would capture ones in a string I believe. Oh maybe it wouldn't. Let me try it.

---
In reply to: [267960402](https://github.com/Microsoft/vscode-python/pull/4875#discussion_r267960402) [](ancestors = 267960402)

Reply:
No it does work with strings.

But makes it more difficult to parse 'from x import b' statements. I have to second parse to get rid of the 'import' value.  I think I'll just leave the one I have there.

---
In reply to: [267961343](https://github.com/Microsoft/vscode-python/pull/4875#discussion_r267961343) [](ancestors = 267961343,267960402)